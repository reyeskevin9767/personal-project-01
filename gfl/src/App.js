import React, { useState, useEffect } from 'react';
import { dolls } from 'girlsfrontline-core';
import SelectStats from './components/SelectStats';
import DollStats from './components/DollStats';
// import DollDetails from './components/DollDetails';
// import InDepthDetails from './gfcore.json';

const App = () => {
  //* Store the stats and details about T-Doll
  const [dollStats, setDollStats] = useState({});
  const [dollDetails, setDollDetails] = useState({});

  //* Create an array contaning all T-Dolls names
  const allDollsNames = [];

  for (let doll in dolls) {
    allDollsNames.push(dolls[doll].codename);
  }

  //* After first render, call onFormSubmit to have a default T-Doll
  useEffect(() => {
    onFormSubmit('M9', 100, 50, 5);
  }, []);

  //* Function -> Pass down to child, finds the specific T-doll
  //* along with calculating stats and getting all T-Doll's details
  const onFormSubmit = (name, level, friendship, dummyLink) => {
    const doll = dolls.find(({ codename }) => codename === name);

    doll.level = level;
    doll.dummyLink = dummyLink;
    doll.favor = friendship;

    setDollDetails(doll);
    setDollStats(doll.stats);
  };

  console.log(dollStats);

  return (
    <div>
      <h1>Found Out the Stats</h1>
      <SelectStats
        onFormSubmit={onFormSubmit}
        allDollsNames={allDollsNames.sort()}
      />
      <DollStats dollStats={dollStats} dummyLink={dollDetails.dummyLink} />
      {/* <DollDetails dollDetails={dollDetails} inDepthDetails={InDepthDetails}/> */}
    </div>
  );
};

export default App;
