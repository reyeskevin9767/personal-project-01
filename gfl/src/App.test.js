import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

// Test App
describe('App Starting Up', () => {
  // Test if app runs
  it('renders without crashing', () => {
    shallow(<App />);
  });

  // Test if app matches snapshot
  it('should render correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });


  // Test if title is correctly showing
  it('renders app title', () => {
    const wrapper = shallow(<App />);
    const welcome = <h1>Found Out the Stats</h1>;
    expect(wrapper.contains(welcome)).toEqual(true);
  });



});
