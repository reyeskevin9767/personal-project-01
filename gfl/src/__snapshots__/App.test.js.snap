// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Starting Up should render correctly 1`] = `ShallowWrapper {}`;
